gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/BloodPressureMapper.java

```diff
@@ -59,71 +53,56 @@
-                ? CompoundStatementUtil.extractResourcesFromCompound(component4.getCompoundStatement(),
-                    RCMRMT030101UK04Component02::hasCompoundStatement, RCMRMT030101UK04Component02::getCompoundStatement)
-                .stream()
-                .map(RCMRMT030101UK04CompoundStatement.class::cast)
-                : Stream.empty()
-        );
+    private Observation mapObservation(RCMRMT030101UK04EhrExtract ehrExtract, RCMRMT030101UK04EhrComposition ehrComposition,
+        RCMRMT030101UK04CompoundStatement compoundStatement, Patient patient, List<Encounter> encounters, String practiseCode) {
+        var observationStatements = getObservationStatementsFromCompoundStatement(compoundStatement);
+        var id = compoundStatement.getId().get(0);
+
+        Observation observation = new Observation()
+            .addIdentifier(buildIdentifier(id.getRoot(), practiseCode))
+            .setStatus(ObservationStatus.FINAL)
+            .setCode(getCode(compoundStatement.getCode()))
+            .setComponent(getComponent(observationStatements))
+            .setComment(
+                getComment(observationStatements, getNarrativeStatementsFromCompoundStatement(compoundStatement)))
+            .setSubject(new Reference(patient))
+            .setIssuedElement(getIssued(
+                ehrExtract,
+                ehrComposition))
+            .addPerformer(getParticipantReference(
+                compoundStatement.getParticipant(),
+                ehrComposition));
+
+        observation.setId(id.getRoot());
+        observation.getMeta().getProfile().add(new UriType(META_PROFILE));
+
+        addEffective(observation, getEffective(compoundStatement.getEffectiveTime(), compoundStatement.getAvailabilityTime()));
+        addContextToObservation(observation, encounters, ehrComposition);
+
+        return observation;
     }
 
-    private List<RCMRMT030101UK04EhrComposition> getCompositionsContainingCompoundStatement(RCMRMT030101UK04EhrExtract ehrExtract) {
-        return ehrExtract.getComponent().stream()
-            .flatMap(component -> component.getEhrFolder().getComponent().stream())
-            .map(RCMRMT030101UK04Component3::getEhrComposition)
-            .filter(ehrComposition -> ehrComposition.getComponent()
-                .stream()
-                .map(RCMRMT030101UK04Component4::getCompoundStatement)
-                .anyMatch(Objects::nonNull))
-            .toList();
+    private boolean containsValidBloodPressureTriple(RCMRMT030101UK04CompoundStatement compoundStatement) {
+        var observationStatements = getObservationStatementsFromCompoundStatement(compoundStatement);
+
+        if (observationStatements.size() == 2) {
+            return BloodPressureValidatorUtil.validateBloodPressureTriple(compoundStatement.getCode().getCode(),
+                observationStatements.get(0).getCode().getCode(), observationStatements.get(1).getCode().getCode());
+        }
+
+        return false;
```

Comment:
might be wrong, but this might be callable from the BloodPressureUtil class now

Reply:
will take a look.

Reply:
it is a slightly different function it looks like but I will move it.